gstvstaudioprocessor.cpp

```diff
@@ -1294,6 +1306,15 @@ gst_vst_audio_processor_register(GstPlugin * plugin)
       GST_DEBUG("\t Class: %s, category: %s", class_info.name().c_str(),
                 class_info.category().c_str());
 
+      GTypeQuery type_query;
+      g_type_query(gst_vst_audio_processor_get_type(), &type_query);
+      auto type_name = create_type_name(type_query.type_name, class_info.name().c_str());
```

Comment:
While type_name is `auto`, it's still a `gchar *` and needs to be freed unfortunately. Either add the `g_free()` below in the various places, or change `create_type_name()` (and probably `create_element_name()`?) to return a `std::string` instead.

Generally it might make sense to use RAII C++ types for things where it makes sense here, and maybe also place more C pointers into smart pointers.

Reply:
Done, this function might deserve some factoring out as well but that's another concern :)